### ISSUE TITLE
refresh our use of node-http-proxy on awsbox AMI to use the latest, http-proxy@0.10.3 when using nodejs 0.10.x

### ISSUE THREAD

**Issue by jrgm** [mozilla/persona#3851]
*\* TL;DR http-proxy@0.8.0 does not work correctly when run with nodejs >0.10.x **

I started writing this up as an issue for https://github.com/mozilla/123done, but since it can affect other browserid-related repos using awsbox deploys, that I'd put the issue in a central place.

I was seeing problems when running automation on 123done in parallel after its update this week to run on nodejs 0.10.7. HTTP requests would mysteriously hang, causing test failures. Anyways, in hindsight, pretty simple reason. 

(/cc @stephendonner in case he's noticing more flakiness recently with bidpom tests for webqa).

STR

1) load http://123done.org in Firefox or Chrome (probably any browser)

2) ssh ec2-user@123done.org

3) In the shell run this command:

```
while [ 1 ]; do date; sudo netstat -tanp | egrep -v 'LISTEN|redis-server|Active|Recv-Q|/sshd|:6379' |\
    awk '{ print $6 }' | sort | uniq -c | sort -k2; sleep 10; done
```

4) Simply hit reload in the browser and let the loads complete. Repeat...

5) After maybe 8 or 10 reloads the browser will spin waiting on network requests (check firebug, etc.)

At the point, you will something like this in the terminal window:

```
Fri Aug 30 22:08:28 UTC 2013
    206 ESTABLISHED
     47 TIME_WAIT
```

206 === 6 sockets from the browser to the proxy, 100 sockets from the proxy to 123done, and 100 sockets on 123done. node-http-proxy by default sets maxSockets to 100.

I duplicated this on an awsbox push of 123done code, and then downgraded nodejs to 0.8.24. With that version of nodejs, I can't reproduce the same issue. So it's an issue with the upgrade to nodejs 0.10.x.

Anyways, the problem is that connections in the proxy are not being shut down in some cases, so the connection pool gets saturated with two-minute long connections that have no real work to do anymore. 

I git-bisected this back to this commit in node-http-proxy@v0.9.0 - https://github.com/nodejitsu/node-http-proxy/commit/28a3b0a6f04197665d95f7574efb4289c259a647, in particular the return statement that is apparently preventing some necessary cleanup when run in nodejs v0.10.x.

Backporting that commit to node-http-proxy@v0.8.0 also works, but perhaps we should just refresh our use of node-http-proxy on awsbox AMI to use the latest, http-proxy@0.10.3, which appears to work for me (when using nodejs 0.10.x).

**Comment by jrgm** [comment]
I did the following on 123done.org

1) sudo su - proxy
2) edit ~proxy/package.json to use http-proxy@0.10.3
3) mv ~proxy/node_modules ~proxy/node_modules-0.8.0
4) npm install
5) forever restartall

So 123done.org is now running with http-proxy@0.10.3. It works for me manually, and the problem with
hitting maxsocket when running automation-tests is gone.

Leaving open for a more formal fix to awsbox ami's when switching to node version 0.10.x.

**Comment by jrgm** [comment]
Oops. Ephemeral's have been switched to an AMI that uses node 0.10.18.

We need to get the AMI updated to using http-proxy@0.10.3 for ~proxy. /cc @lloyd

**Comment by jrgm** [comment]
Hmm, it looks like this change is in ami-4b1c5722 which is what new boxes are being built with. Did you fix this @lloyd? Close if so (and thanks).

**Comment by jrgm** [comment]
Hrm, it seems that there's something else going on - see https://github.com/mozilla/browserid/issues/3895 (after fixing the vows issue).

I'll file a new issue to not confuse things here. (But it's related).
